### ISSUE TITLE
cant close a tab with no changes (but filter is set)

### ISSUE THREAD

**Issue by linojon** [prisma/studio#727]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by madebysid** [comment]
Hey @linojon, IIUC, adding a filter to a tab marks it as "dirty", and you expect it to not be that way?
If so, then this is expected. We mark it so so that you don't accidentally close tabs that might contain manual input.

Let me know if I misunderstood!

**Comment by linojon** [comment]
Thanks. I’ve gotten used to it now. But my expectation was that it wouldn’t be dirty unless I had unsaved changes. New tabs are opened with filters when browsing relation connections, and now closing those tabs requires several extra clicks which is annoying.

**Comment by linojon** [comment]
Its a pain to close tabs when its "dirty", especially when theres alot open eg from navigating relations. 

The easiest solution is allow closing a tab when you click to close it, without the Discard Changes dialog box when there are no pending "Save N change". I see no reason to consider the tab as dirty just because the view has changed eg theres a filter or sort etc set. 

But if you insist that there must be a Discard Changes box in those cases, heres some suggestions for UI improvement

* Allow keyboard enter key to Discard on the Discard Changes? dialog
* Right-click on tab > context menu > Close & Discard
* Setting icon > Close & Discard All (sometimes i just want a clean slate)

thanks

**Comment by AronBe** [comment]
This is really hard to close, can't we have close all tabs util? :)

**Comment by petradonka** [comment]
We're working on and will soon be releasing some improvements to closing tabs that will better differentiate between actual unsaved changes and filters that were set, and helping you close tabs and discard any of those changes.
Even then, I still see the value in some of the solutions outlined in this issue, e.g. a close all feature. I'll bring this up internally and put it on our long-term roadmap.

**Comment by mttias** [comment]
Is this resolved? This would help me a lot.

**Comment by nate-fur** [comment]
I have to agree that the experience of closing tabs that have filters applied is a bit cumbersome. One thing that would smooth out this experience would be to have the "Close filtered view" button _focused_ when the dialog opens. This would allow using the enter button on a keyboard to close the view rather than having to traverse back and forth between that button and the several filtered tabs that I need to close in a row.
![Screenshot 2023-12-08 at 7 49 35 AM](https://github.com/prisma/studio/assets/37522274/f6ae9093-0293-40da-9210-9ad600904fb0)

Two other options to help smooth out this experience would be
1. Not show a confirmation dialog when closing a filtered view. (I haven't found the confirmation to be useful for filtered views, only for unsaved changes)
2. Implement some sort of "close all tabs" functionality.

**Comment by banquetteapp** [comment]
Any updates on this?

**Comment by igalklebanov** [comment]
Hey :wave:

Prisma ORM 7 introduces a brand new standalone [Studio](https://www.prisma.io/docs/orm/tools/prisma-studio), built from the ground up to work everywhere, with or without Prisma ORM. Consequently, all "old Studio" issues are being closed as there won't be any further work on it. :mega:

The new Studio is also available in the Prisma VS Code extension and in the Prisma Console application. :electric_plug:

Help shape, improve and fix the new Studio with us in this repository by submitting bug reports and feature requests. :construction_worker:
